tests/test_record_requests_for_all_methods.py

```diff
@@ -45,6 +45,48 @@ def test_write(self, cassette: Cassette):
         Path(cassette.storage_file).unlink()
 
 
+@record_requests_for_all_methods()
+class DecoratorRecordRequestForAllMethodsCassetteSetupTearDown(unittest.TestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.count_cassette_setup_triggered = 0
+        self.count_cassette_teardown_triggered = 0
+
+    def tearDown(self) -> None:
+        super().tearDown()
+        self.assertEqual(self.count_cassette_setup_triggered, 1)
+        self.assertEqual(self.count_cassette_teardown_triggered, 1)
+
+    def cassette_setup(self, cassette: Cassette):
+        self.assertIsNotNone(cassette)
+        self.count_cassette_setup_triggered += 1
+
+    def cassette_teardown(self, cassette: Cassette):
+        self.assertIsNotNone(cassette)
+        self.count_cassette_teardown_triggered += 1
+
+    def use_requests(self):
+        response = requests.get("http://example.com")
+        self.assertIn("This domain is for use", response.text)
+
+    def test_read(self, cassette: Cassette):
+        self.assertIn("requests.sessions", cassette.storage_object)
+        # move the following call to the beginning when regenerating
+        self.use_requests()
+
+    def test_write(self, cassette: Cassette):
+        self.assertIsNotNone(cassette)
+
+        self.use_requests()
+
```

Comment:
please add there something like :
```
        self.assertEqual(self.count_cassette_setup_triggered, 1)
        self.assertEqual(self.count_cassette_teardown_triggered, 0)
```
to ensure that teardown is not executed at the beginning.

Reply:
good catch, fixed!